plugins/catalog-backend-module-bitbucket-server/src/providers/BitbucketServerEntityProvider.ts

```diff
@@ -37,6 +38,14 @@ import {
   BitbucketServerLocationParser,
   defaultBitbucketServerLocationParser,
 } from './BitbucketServerLocationParser';
+import { Events } from '../lib/index';
+import { EventParams } from '@backstage/plugin-events-node';
+import { CatalogApi } from '@backstage/catalog-client';
+import { TokenManager } from '@backstage/backend-common';
+import { EventBroker } from '@backstage/plugin-events-node';
+
+const DEFAULT_BRANCH = ['master', 'main'];
+const TOPIC_REPO_PUSH = 'bitbucketServer.repo:refs_changed';
```

Comment:
I realized for other providers that you may be interested in a few other events as well (e.g. if the repo gets renamed or archived, etc.).

You might want to check whether there are other relevant events. Maybe for a follow-up.

Reply:
Understood, however, I was just intending to catch the Bitbucket Server plugin up with Bitbucket Cloud as of this PR as the "push" event in Bitbucket Cloud (similar to "refs_changed") is the only one included in that plugin as of right now. Definitely a great idea for a follow-up!

Reply:
I was the one who added the event support there (it was the trigger to work on the event support overall for me). However, I realized some uncovered use cases while we had this in production use. Definitely not required for this PR. :)